src/dotty/tools/dotc/core/Types.scala

```diff
@@ -927,15 +927,18 @@ object Types {
     def narrow(implicit ctx: Context): TermRef =
       TermRef(NoPrefix, ctx.newSkolem(this))
 
-    /** Useful for diagnsotics: The underlying type if this type is a type proxy,
+    /** Useful for diagnostics: The underlying type if this type is a type proxy,
      *  otherwise NoType
      */
     def underlyingIfProxy(implicit ctx: Context) = this match {
       case this1: TypeProxy => this1.underlying
       case _ => NoType
     }
 
-    // ----- Normalizing typerefs over refined types ----------------------------
+	/** If this is a FunProto or PolyProto, WildcardType, otherwise this */
```

Comment:
I'm also not entirely sure why using `WildcardType` does not break type inference in some cases, maybe this could be explained here